manager/catalog_manager.go

```diff
@@ -47,16 +58,90 @@ var (
 	//ValidationMode is used to determine if code is just checking Yaml syntax
 	ValidationMode bool
 
-	catalogURL arrayFlags
+	catalogURL     arrayFlags
+	commandLineURL arrayFlags
+
+	//URLBranchMap aps repo url to branch
+	URLBranchMap map[string]string
+
+	reloadChan = make(chan chan error)
 )
 
 //CatalogRootDir is the root folder under which all catalogs are cloned
 const CatalogRootDir string = "./DATA/"
 
-//SetEnv parses the command line args and sets the necessary variables
-func SetEnv() {
+//WatchSignals handles SIGHUP
+func WatchSignals() {
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, syscall.SIGHUP)
+
+	go func() {
+		for _ = range c {
+			log.Info("Received HUP signal")
+			SetEnv()
+			go Init()
+		}
+	}()
+}
+
+//GetCommandLine parses the command line args
+func GetCommandLine() {
 	flag.Var(&catalogURL, "catalogUrl", "git repo url in the form repo_id=repo_url. Specify the flag multiple times for multiple repos")
+
 	flag.Parse()
+	commandLineURL = catalogURL
+	SetEnv()
+}
+
+//SetEnv parses the command line args and sets the necessary variables
+func SetEnv() {
+
+	catalogURL = catalogURL[:0]
+	catalogURL = commandLineURL
+
+	var catalogObject []CatalogInput
+	var catalogObjectJSON []CatalogInput
+	var catalogObjectNonEmptyJSON []CatalogInput
+	var catalogObjectCommandLine []CatalogInput
+	URLBranchMap = make(map[string]string)
+
+	//NEW CODE
+	// If catalog provided through command line
+	if len(catalogURL) > 0 {
+		for i := 0; i < len(catalogURL); i++ {
+			catalogObj := CatalogInput{}
+			catalogObj.RepoURL = catalogURL[i]
+			catalogObj.Branch = "master"
+			catalogObjectCommandLine = append(catalogObjectCommandLine, catalogObj)
+			URLBranchMap[catalogObj.RepoURL] = catalogObj.Branch
+		}
+		catalogObject = append(catalogObject, catalogObjectCommandLine...)
+	}
+
+	libraryContent, err := ioutil.ReadFile("./repo.json")
```

Comment:
The [path to and] name of the file should be a command line flag, in real-world use it will live in something like `/var/lib/rancher/etc/catalog/config.json`.